android/app/src/main/java/com/mattermost/rnbeta/NotificationsLifecycleFacade.java

```diff
@@ -10,18 +16,67 @@
 import com.wix.reactnativenotifications.core.AppLifecycleFacade;
 
 import java.util.Set;
+import 	android.util.ArraySet;
 import java.util.concurrent.CopyOnWriteArraySet;
+import android.view.WindowManager.LayoutParams;
+
 
 public class NotificationsLifecycleFacade extends ActivityCallbacks implements AppLifecycleFacade {
 
     private static final String TAG = NotificationsLifecycleFacade.class.getSimpleName();
+    private Bundle managedConfig = null;
 
     private Activity mVisibleActivity;
     private Set<AppVisibilityListener> mListeners = new CopyOnWriteArraySet<>();
 
+    private final IntentFilter restrictionsFilter =
+            new IntentFilter(Intent.ACTION_APPLICATION_RESTRICTIONS_CHANGED);
+
+    private final BroadcastReceiver restrictionsReceiver = new BroadcastReceiver() {
+        @Override public void onReceive(Context context, Intent intent) {
+
+            if (mVisibleActivity != null) {
+                // Get the current configuration bundle
+                RestrictionsManager myRestrictionsMgr =
+                    (RestrictionsManager) mVisibleActivity
+                            .getSystemService(Context.RESTRICTIONS_SERVICE);
+                managedConfig = myRestrictionsMgr.getApplicationRestrictions();
+
+                // Check current configuration settings, change your app's UI and
+                // functionality as necessary.
+                Log.i("ReactNative", "Managed Configuration Changed");
+                MattermostManagedModule.instance.sendConfigChanged(managedConfig);
```

Comment:
Same question as above - is it possible to pass an instance of `MattermostManagedModule` into this object at creation time so that you can avoid static access?